src/WebJobs.Script.WebHost/Diagnostics/ILoggingBuilderExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Logging
+{
+    public static class ILoggingBuilderExtensions
+    {
+        public static void AddWebJobsSystem(this ILoggingBuilder builder)
+        {
+            // Do not filter the SystemLogger
+            builder.Services.AddSingleton<ILoggerProvider, SystemLoggerProvider>();
+            builder.AddFilter<SystemLoggerProvider>(_ => true);
+        }
+
+        public static void AddDeferred(this ILoggingBuilder builder)
+        {
+            // Do not filter this. It will be filtered internally.
+            builder.Services.AddSingleton<IScriptWebHostEnvironment, ScriptWebHostEnvironment>();
```

Comment:
The call to `Build()` requests an ILogger before this is set up, but it's required by the DeferredLogger so we can ignore StandbyMode logs. So moving it up here.